src/components/Buttons/ButtonsAdmin/ButtonsAdmin.jsx

```diff
@@ -1,28 +1,164 @@
+    } else {
+      toggleError();
+    }
+  };
+
+  toggleModalEditHandler = () => {
+    this.setState((prevState) => ({ isEditModalOpen: !prevState.isEditModalOpen }));
+  };
+
+  toggleModalDeleteHandler = () => {
+    this.setState((prevState) => ({ isDeleteModalOpen: !prevState.isDeleteModalOpen }));
+  };
+
+  showUserDataHandler = async () => {
+    await this.getUserData();
+    this.toggleModalEditHandler();
+  };
+
+  render() {
+    const { id, setUsersHandler, userName, readMode, toggleError } = this.props;
+    const { isDeleteModalOpen, isEditModalOpen, userData } = this.state;
+
+    return (
+      <AuthContext.Consumer>
+        {({ role }) => {
+          return readMode ? (
+            <>
+              <div className={style.name} role='none' onClick={this.showUserDataHandler}>
+                {userName}
+              </div>
+              {isEditModalOpen && (
+                <MyModal
+                  title={userName}
+                  isModalOpen={isEditModalOpen}
+                  toggleModalHandler={this.toggleModalEditHandler}
+                >
+                  <CreateMemberForm
+                    toggleError={toggleError}
+                    isReadOnlyMode
+                    isEditMode
+                    id={id}
+                    userData={userData}
+                    toggleModalHandler={this.toggleModalEditHandler}
+                    setUsersHandler={setUsersHandler}
+                  />
+                </MyModal>
+              )}
+            </>
+          ) : (
+            <>
+              <NavLink to={`/tasks/${id}`}>
+                <Button title={BUTTONS_NAMES.tasks} />
+              </NavLink>
+              <NavLink to={`/progress/${id}`}>
+                <Button title={BUTTONS_NAMES.progress} />
+              </NavLink>
+              {role === USER_ROLES.admin && (
+                <>
+                  <Button
+                    title={BUTTONS_NAMES.edit}
+                    stylingType={BUTTONS_TYPES.typeEdit}
+                    onClick={this.showUserDataHandler}
+                  />
+                  <Button
+                    title={BUTTONS_NAMES.delete}
+                    stylingType={BUTTONS_TYPES.typeDelete}
+                    onClick={this.toggleModalDeleteHandler}
+                  />
+                </>
+              )}
+
+              {isDeleteModalOpen && (
+                <MyModal
+                  title='Delete member'
+                  isModalOpen={isDeleteModalOpen}
+                  toggleModalHandler={this.toggleModalDeleteHandler}
+                >
+                  <DeleteForm
+                    toggleError={toggleError}
+                    item='member'
+                    deleteHandler={this.deleteUserHandler}
+                    toggleModalHandler={this.toggleModalDeleteHandler}
+                  />
+                </MyModal>
+              )}
+              {isEditModalOpen && (
+                <MyModal
```

Comment:
Could you give it more appropriate and accurate name ?

Reply:
now `<ModalWindow>`